src/build_loop.rs

```diff
@@ -96,6 +99,7 @@ impl<'a> BuildLoop<'a> {
         let (tx, rx) = channel();
         let run_result = builder::run(tx, &self.project.nix_file, &self.project.cas)?;
 
+        self.linter.check(&run_result.referenced_paths);
```

Comment:
This is a good start, though it matches a bit too much.

If you look at

https://github.com/target/lorri/blob/8c5a3fbd0dbeb6d54b6a9ea3eac10ea03e0f770f/src/builder.rs#L154-L184

You see that `paths` (which turns into `referenced_paths` at the end) is constructed from multiple `LogDatum` sources

https://github.com/target/lorri/blob/8c5a3fbd0dbeb6d54b6a9ea3eac10ea03e0f770f/src/builder.rs#L298-L311

We only really need the `CopiedSource` case, because that’s where the user can make a mistake (in copying to much because of a wrong `builtins.filterSource` filter).

So we should add a new field to the return of `instrumented_instantiation`, which returns the copied sources and then pipe that through `builder::run()` and out to `build_loop`.

I can help with that if that’s too much overhead.